File: MLD701/code/runNB.m

clear;

fprintf('Loading training data...\n');
trainData = load('../data/train.data');
trainLabels = load('../data/train.label');

fprintf('Loading testing data...\n');
testData = load('../data/test.data');
testLabels = load('../data/test.label');

fprintf('Computing log prior...\n');
lp = logPrior(trainLabels);
csvwrite('logPrior.csv', lp);

fprintf('Computing log likelihood...\n');
llh = logLikelihood(trainData, trainLabels);
csvwrite('logLikelihood.csv', llh);

fprintf('Classifying using Naive Bayes...\n');
pred = naiveBayesClassify(lp, llh, testData);
csvwrite('nbResults.csv', pred);

res = (pred == testLabels);
accuracy = numel(res(res == 1)) / numel(res) * 100;
fprintf('Accuracy is %0.2f\n', accuracy);

File: MLD701/HW1/code/runNB.jl

include("naivebayes.jl")

print("Loading training data...\n")
trainData = readdlm("../data/train.data", ' ', Int)
trainLabels = readdlm("../data/train.label", ' ', Int)

print("Loading testing data...\n")
testData = readdlm("../data/test.data", ' ', Int)
testLabels = readdlm("../data/test.label", ' ', Int)

print("Computing log prior...\n")
lp = MLNaiveBayes.logPrior(trainLabels)
writecsv("logPrior.csv", lp)

print("Computing log likelihood...\n")
llh = MLNaiveBayes.logLikelihood(trainData, trainLabels)
writecsv("logLikelihood.csv", llh)

print("Classifying using Naive Bayes...\n")
pred = MLNaiveBayes.naiveBayesClassify(lp, llh, testData)
writecsv("nbResults.csv", pred)

res = (pred .== testLabels)
accuracy = length(res[res]) / length(res) * 100
@printf("Accuracy is %0.2f\n", accuracy)

